cirq/google/api/v2/program.proto

```diff
@@ -198,3 +216,81 @@ message ArgFunction {
   // The arguments to the function.
   repeated Arg args = 2;
 }
+
+// An operation that describes a circuit.
+message CircuitOperation {
+  // The index of the circuit in the top-level constant table.
+  int32 circuit_constant_index = 1;
+
+  // Specifier for repetitions of the circuit, including number of reps and
+  // repetition IDs.
+  RepetitionSpec repetition_spec = 2;
+
+  // Map of qubits in the circuit to the qubits used in this operation.
+  QubitMapping qubit_map = 3;
```

Comment:
Why do we need a qubit mapping and measurement key mapping in the serialized form?

Why can't we just map the qubits and keys while serializing?  Is this because we want to re-use the circuit in the index multiple times with different mappings?

Reply:
> Is this because we want to re-use the circuit in the index multiple times with different mappings?

This is correct. Consider the case of three `CircuitOperation`s {A, B, C}, each with the same underlying circuit M but with different qubit mappings. Serializing the maps allows these `CircuitOperation`s to share a single circuit.

A real-world example of this is two-qubit XEB, where we want to run a specific circuit (XEB) on a whole bunch of qubits.